r/R/type.R

```diff
@@ -665,13 +694,16 @@ list_of <- function(type) list__(type)
 LargeListType <- R6Class("LargeListType",
   inherit = NestedType,
   public = list(
-    code = function() {
-      call2("large_list_of", self$value_type$code())
+    code = function(explicit_pkg_name = FALSE) {
+      call2(private$call_name(), self$value_type$code(explicit_pkg_name), .ns = get_pkg_ns(explicit_pkg_name))
     }
   ),
   active = list(
     value_field = function() LargeListType__value_field(self),
     value_type = function() LargeListType__value_type(self)
+  ),
+  private = list(
+    call_name = function() "large_list_of"
```

Comment:
The code feels more skimmable to me with the call name written directly inside the `code()` method instead of pulled from a private field, so I'd lean towards leaving it as-is, unless there's another strong reason to add this here?

Reply:
Separating out the `call_name` came from a previous implementation where I was doing much more complicated things, before I discovered the `.ns` argument in `call2`. I agree with you that we should revert to the previous setup.